jscomp/bin/whole_compiler.ml

```diff
@@ -81072,6 +81072,13 @@ type name =
   | Name_top  of Ident.t
   | Name_non_top of Ident.t
 
+let rec process_comments (el:J.expression list) = match el with 
+| [] -> []
+| e::r -> let new_e = 
+            if e.comment = Some "::" || e.comment = Some "[]"
+            then {J.expression_desc = e.expression_desc; J.comment = None} 
+            else e in
+          new_e::(process_comments r)
 
```

Comment:
I would imagine the output would be 

```js
var x = /* Nested :: */ [ 1 ,
                                        [2, /* [] */ 0 ] ]
```
This would be strictly better, and still correct